libraries/AP_SerialManager/AP_SerialManager.cpp

```diff
@@ -482,68 +531,264 @@ void AP_SerialManager::init()
+#if !HAL_GENERATOR_ENABLED
+                    protocol_not_compiled_in_error("Generator", i);
+#endif
+                    break;
+                case SerialProtocol_Winch:
+#if AP_WINCH_ENABLED
+                    uart->begin(state[i].baudrate());
+#else
+                    protocol_not_compiled_in_error("Winch", i);
+#endif
+                    break;
+                case SerialProtocol_AirSpeed:
+#if AP_AIRSPEED_ENABLED
+                    uart->begin(state[i].baudrate());
+#else
+                    protocol_not_compiled_in_error("AirSpeed", i);
+#endif
+                    break;
+                case SerialProtocol_ADSB:
+#if HAL_ADSB_ENABLED
+                    uart->begin(state[i].baudrate());
+#else
+                    protocol_not_compiled_in_error("ADSB", i);
+#endif
+                    break;
+                case SerialProtocol_SmartAudio:
+#if AP_SMARTAUDIO_ENABLED
+                    check_no_duplicates_error(SerialProtocol_SmartAudio, i);
+                    uart->begin(state[i].baudrate());
+#else
+                    protocol_not_compiled_in_error("SmartAudio", i);
+#endif
+                    break;
+                case SerialProtocol_FETtecOneWire:
+#if AP_FETTEC_ONEWIRE_ENABLED
+                    uart->begin(state[i].baudrate());
+#else
+                    protocol_not_compiled_in_error("FETtecOneWire", i);
+#endif
+                    break;
+                case SerialProtocol_Torqeedo:
+#if HAL_TORQEEDO_ENABLED
+                    uart->begin(state[i].baudrate());
+#else
+                    protocol_not_compiled_in_error("Torqeedo", i);
+#endif
+                    break;
+                case SerialProtocol_AIS:
+#if AP_AIS_ENABLED
+                    uart->begin(state[i].baudrate());
+#else
+                    protocol_not_compiled_in_error("AIS", i);
+#endif
+                    break;
+                case SerialProtocol_CoDevESC:
+                    protocol_not_compiled_in_error("CoDevESC", i);
+                    break;
+                case SerialProtocol_Tramp:
+#if AP_TRAMP_ENABLED
+                    check_no_duplicates_error(SerialProtocol_Tramp, i);
+                    uart->begin(state[i].baudrate());
+#else
+                    protocol_not_compiled_in_error("Tramp", i);
+#endif
+                    break;
+                case SerialProtocol_DDS_XRCE:
+#if AP_DDS_ENABLED
+                    uart->begin(state[i].baudrate());
+#else
+                    protocol_not_compiled_in_error("DDS_XRCE", i);
+#endif
+                    break;
+                case SerialProtocol_AHRS:
+                case SerialProtocol_WindVane:
                 default:
                     uart->begin(state[i].baudrate());
             }
         }
     }
 }
 
+void AP_SerialManager::protocol_not_compiled_in_error(const char* tag, uint8_t serial) const
+{
+    AP_BoardConfig::config_error("SerialProtocol_%s required on SERIAL%u not compiled into firmware", tag, serial);
```

Comment:
config_error will stop the boot, and is being done when serial manager hasn't be initialised, so STATUSTEXT not available, so I think this would just appear as a bricked board on MissionPlanner

Reply:
Fixed, thanks